ee/agent/startupsettings/writer.go

```diff
@@ -63,6 +65,18 @@ func (s *startupSettingsWriter) setFlags() error {
 	}
 	updatedFlags["use_tuf_autoupdater"] = "enabled" // Hardcode for backwards compatibility circa v1.5.3
 
+	// Set flags will only be called when a flag value changes. The osquery config that contains the atc config
+	// comes in via osquery extension. So a new config will not trigger a re-write.
+	atcConfig, err := s.extractAutoTableConstructionConfig()
```

Comment:
Is it worth figuring out the plumbing to have this fire when we get a new osquery config? This would likely mean that launcher interactive will not pick up the ATCs until the first launcher start up after getting osq config.

Reply:
I'm not sure I understand... Is this only called on startup? Which means the sqlite conf file may lag weeks to months?

Reply:
Yes, it's possible. I'm trying to find a clever way to plumb this through .... we could treat config like a flag in knapsack and use it's observer pattern or just find a way to have the extension call startupsettings.OpenWriter when the config updates.

Reply:
mmm, that's an interesting thought. If we ever decide we need to restart osquery on a config change, would it leverage that?

Reply:
I doubt we would ever leverage that when launcher (daemon) is restarting osquery, since launcher (daemon) can just read the config out of bolt db.

Reply:
Right -- that extension would read it from boltdb. But we still need something like that to trigger the restart, don't we?

Reply:
I've updated this to have the extension just open the startup settings writer to trigger the write when it gets a new config. I see what you mean about restarting on a new config, as discussed, maybe an issue, but out of scope for this PR.

Reply:
Agree it's out of scope for this PR. Mostly mentioning it in case it chanses the direction you want to go in